lib/centoswebpanel_installer.php

```diff
@@ -0,0 +1,203 @@
+            if (trim(strtolower($value['name'])) == trim(strtolower($installationScript))) {
+                $sid = $key;
+                break;
+            }
+        }
+
+        // Did we find the Script ?
+        if (empty($sid)) {
+            $this->Input->setErrors([
+                'script_id' => [
+                    'invalid' => 'Could not determine the script to be installed.'
+                    . ' Please make sure the script name is correct. Script Name : ' . $installationScript
+                ]
+            ]);
+            return;
+        }
+
+        // Install the script
+        $response = $this->scriptInstallRequest(
+            $sid,
+            $loginResponse->redirect_url,
+            $data
+        );
+
+
+        if ('installed' == strtolower($response)) {
+            return true;
+        } else {
+            $messages = unserialize($response);
+            $this->Input->setErrors([
+                'script_id' => [
+                    'invalid' => 'Could not install script: ' . ($messages ? $messages[0] : $response)
+                ]
+            ]);
+            return false;
+        }
+    }
+
+    /**
+     * Send a request to the CentOS WebPanel.
+     *
+     * @param array $post The parameters to include in the request
+     * @param string $url Specifies the url to invoke
+     * @param string $method Http request method (GET, DELETE, POST)
+     * @return string An json formatted string containing the response
+     */
+    private function post($post, $url, $method = 'GET')
+    {
+        $ch = curl_init();
+
+        // Set the request method and parameters
+        switch (strtoupper($method)) {
+            case 'GET':
+            case 'DELETE':
+                $url .= empty($post) ? '' : '?' . http_build_query($post);
+                break;
+            case 'POST':
+                curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "POST");
+                curl_setopt($ch, CURLOPT_POST, 1);
+            default:
+                curl_setopt($ch, CURLOPT_POSTFIELDS, http_build_query($post));
+                break;
+        }
+
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch, CURLOPT_VERBOSE, 0);
+
+        // Turn off the server and peer verification (TrustManager Concept).
+        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+        curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false);
+
+        // Create new session cookies
+        curl_setopt($ch, CURLOPT_COOKIESESSION, true);
+
+        // Check the Header
+        curl_setopt($ch, CURLOPT_HEADER, 1);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+
+        // Get response from the server.
+        $response = curl_exec($ch);
+
+        $this->setCookie($response);
+
+
+        $error = curl_error($ch);
+        if ($error !== '') {
+            $this->Input->setErrors([
+                'login' => [
+                    'invalid' => 'Could not login to the remote server. cURL Error : ' . $error
```

Comment:
I'm not sure we should be showing curl errors to clients. 

Also, can these errors be language definitions? Perhaps errors can be set on the class and the parent can check for errors to display